Src/WindowsServer/WindowsServer.Shared.Tests/AzureInstanceMetadataEndToEndTests.cs

```diff
@@ -62,32 +65,30 @@ public void SpoofedResponseFromAzureIMSDoesntCrash()
         [TestMethod]
         public void AzureImsResponseTooLargeStopsCollection()
         {
-            CancellationTokenSource cts = new CancellationTokenSource();
+            string testPath = "tooLarge";
 
             using (new AzureInstanceMetadataServiceMock(
-                AzureInstanceMetadataEndToEndTests.MockTestUri, 
-                (HttpListenerContext context) =>
-            {
-                HttpListenerResponse response = context.Response;
-                context.Response.StatusCode = 200;
-
-                // Construct a response just like in the positive test but triple it.
-                response.ContentEncoding = Encoding.UTF8;
-
-                // Get a response stream and write the response to it.
-                var jsonStream = this.GetTestMetadataStream();
-                response.ContentLength64 = 3 * (int)jsonStream.Length;
-                context.Response.ContentType = "application/json";
-                jsonStream.WriteTo(context.Response.OutputStream);
-                jsonStream.Position = 0;
-                jsonStream.WriteTo(context.Response.OutputStream);
-                jsonStream.Position = 0;
-                jsonStream.WriteTo(context.Response.OutputStream);
-            }))
+                AzureInstanceMetadataEndToEndTests.MockTestUri,
+                testPath,
+                (response) =>
+                {
+                    response.StatusCode = (int)HttpStatusCode.OK;
+
+                    var jsonStream = this.GetTestMetadataStream();
+                    response.SetContentLength(3 * jsonStream.Length);
```

Comment:
What is the limit above which we ignore the response? Why no make this (1+allowedlimit) ?

Reply:
I've added a test that puts us over the defined max length limit purposefully, instead of the 3x approach.